climt/_components/rrtmg/sw/component.py

```diff
@@ -396,54 +432,131 @@ def array_call(self, state):
         tendencies = initialize_numpy_arrays_with_properties(
             self.tendency_properties, state, self.input_properties
         )
+
+        num_reduced_g_intervals = self.num_reduced_g_intervals
+        mid_levels = state['air_pressure'].shape[0]
+        mcica_properties = {
```

Comment:
Should these be set to zero, or taken as inputs and put into `input_properties`? Will this even work with a cloud water droplet radius of zero?